api/src/models/request_info.rs

```diff
@@ -67,16 +73,40 @@ impl<'r> FromRequest<'r> for RequestInfo {
     type Error = ();
 
     async fn from_request(request: &'r Request<'_>) -> request::Outcome<Self, Self::Error> {
-        let hosts: Vec<_> = request.headers().get("host").collect();
+        // Checking if "x-forwarded-host" header is present
+        let forwarded_host = request.headers().get_one("x-forwarded-host");
+        let host = match forwarded_host {
+            Some(proto) => proto.to_string(),
+            None => request.headers().get_one("host").unwrap().to_string(), // Default to "host" if not present
+        };
 
-        if hosts.len() != 1 {
+        if host == "" {
```

Comment:
Due to `unwarp().to_string()` in line 80 the code my fail with a panic in the server and the server wouldn't respond with bad request. I would remove this if block and adjust the match above like this:

```rust
        let forwarded_host = request.headers().get_one("x-forwarded-host");
        let host = match forwarded_host {
            Some(host) => host.to_string(),
            None => match request.headers().get_one("host") {
                Some(host) => host.to_string(),
                None => {
                    return Outcome::Failure((Status::BadRequest, ()));
                }
            }
        }
```

This has the advantage of saving a little CPU cycles (because there is no string allocation if there is no host header) and it doesn't panic.